testinfra/backend/__init__.py

```diff
@@ -46,11 +46,14 @@ def parse_hostspec(hostspec):
         url = urllib.parse.urlparse(hostspec)
         kw["connection"] = url.scheme
         host = url.netloc
+        path = url.path
+        if path:
+            kw['container'] = path[1:]
```

Comment:
Oh, I see that ``urlparse()`` splitted the part after `/` into this `path` attribute.
So this has never worked.

```
>>> import testinfra
>>> backend = testinfra.get_host('kubectl://pod/c').backend
>>> backend.name
'pod'
>>> backend.container is None
True
```

In this case I think you can just drop this feature of splitting the container and the pod name with `/` and just rely on url querystring.

A good test could be:

````
b = testinfra.get_host('kubectl://pod?namespace=space&container=nginx').backend
assert b.name == 'pod'
assert b.namespace == 'space'
assert b.container == 'nginx'
```

Also I think this deserve some documentation in `doc/source/backend.rst` ;)